flake.nix

```diff
@@ -52,67 +52,74 @@
           }
           // pkgs.lib.optionalAttrs (builtins.readDir path ? "project.ncl") rec {
             nickelOutputs = lib.importNcl path "project.ncl" inputs lockFileContents;
-            packages.default = nickelOutputs.packages.default or {};
+            packages =
+              if nickelOutputs ? packages && nickelOutputs.packages ? default
+              then {
+                default = nickelOutputs.packages.default;
+              }
+              else {};
             devShells = nickelOutputs.shells or {};
           });
-    }
-    // flake-utils.lib.eachDefaultSystem (
-      system: let
-        lib = pkgs.callPackage ./lib/lib.nix {
-          flakeRoot = self.outPath;
-          nickel = inputs.nickel.packages."${system}".nickel-lang-cli;
-        };
-        pkgs = nixpkgs.legacyPackages.${system};
-      in {
-        inherit lib;
+    } [
+      (prev:
+        prev.flake.outputsFromNickel ./. inputs {
+          lockFileContents.organist = "./lib/nix.ncl";
+        })
+      (prev:
+        flake-utils.lib.eachDefaultSystem (
+          system: let
+            lib = pkgs.callPackage ./lib/lib.nix {
+              flakeRoot = self.outPath;
+              nickel = inputs.nickel.packages."${system}".nickel-lang-cli;
+            };
+            pkgs = nixpkgs.legacyPackages.${system};
+          in {
+            inherit lib;
 
-        apps.run-test = let
-          testScript = pkgs.writeShellApplication {
-            name = "test-templates";
-            runtimeInputs = [
-              inputs.nickel.packages."${system}".nickel-lang-cli
-              pkgs.parallel
-              pkgs.gnused
-            ];
-            text = builtins.readFile ./run-test.sh;
-          };
-        in {
-          type = "app";
-          program = pkgs.lib.getExe testScript;
-        };
+            apps =
```

Comment:
Is there a fundamental reason why we can't define the apps in Nickel, or is it just a temporary thing?